utils/sc_extraction_utils.py

```diff
@@ -0,0 +1,164 @@
+    data_df.insert(2, "Metadata_number_of_singlecells", singlecell_column)
+
+    data_df.to_csv(data_path)
+
+def extract_single_cells(
+    single_cell_file: str,
+    linking_cols: dict,
+    platemap_df: pd.DataFrame,
+    output_folder: str,
+    method_name: str,
+    normalize_sc: bool = False,
+    feature_selection_sc: bool = False,
+    norm_feature_select: bool = False,
+):
+    """use Pycytominer SingleCells class to perform single cell extraction, normalization, and feature selection on
+    CellProfiler SQLite files. Normalization and feature selection are optional processes that you can choose
+    to turn on. This function will output all data into individual csv.gz files to a specified output folder. The 
+    individual csv.gz are then updated to contain single cell count metadata.
+
+    Args:
+        single_cell_file (str):
+            string file path to SQLite file (must start with sqlite:///)
+        linking_cols (dict):
+            dictionary with the linking columns between compartments/tables in database
+        platemap_df (pd.DataFrame):
+            dataframe with the platemap metadata to merge with single cells
+        output_folder (str):
+            path to output folder for csv.gz files
+        method_name (str):
+            name of pipeline method used for naming the output files
+        normalize_sc (bool, optional):
+            if set to True, this will perform normalization on the raw merged single cell data (defaults to False)
+        feature_selection_sc (bool, optional):
+            if set to True, this will perform feature extraction on raw merged single cell data (defaults to False)
+        norm_feature_select (bool, optional):
+            if set to True, this will perform normalization on raw merged single cell data and feature extraction on
+            the normalizated data (defaults to False)
+    """
+    # instaniate the SingleCells class
+    sc = cells.SingleCells(
+        sql_file=single_cell_file,
+        compartments=["Per_DilatedNuclei", "Per_TranslocatedNuclei", "Per_Nuclei"],
+        compartment_linking_cols=linking_cols,
+        image_table_name="Per_Image",
+        strata=["Image_Metadata_Well", "Image_Metadata_Plate"],
+        merge_cols=["ImageNumber"],
+        image_cols="ImageNumber",
+        load_image_data=True,
+    )
+
+    # Merge single cells across compartments based on well
+    anno_kwargs = {"join_on": ["Metadata_well", "Image_Metadata_Well"]}
+
+    sc_df = sc.merge_single_cells(
+        platemap=platemap_df,
+        **anno_kwargs,
+    )
+
+    sc_output_file = pathlib.Path(f"{output_folder}/sc_{method_name}.csv.gz")
+
+    # Save level 2 data as a csv
+    output(sc_df, sc_output_file)
+    # add single cell count to the raw single cell data
+    add_sc_count_metadata(sc_output_file)
+
+    # Perform normalization on the raw extracted single cell data
+    if normalize_sc == True:
```

Comment:
Consider using only the variable for the if conditional check here to simplify.

Reply:
I have updated that now!